samples/net/zperf/src/zperf_shell.c

```diff
@@ -218,6 +218,42 @@ int zperf_get_ipv4_addr(const struct shell *shell, char *host,
 	return 0;
 }
 
+struct in_addr *zperf_get_default_if_in4_addr(void)
+{
+#if CONFIG_NET_IPV4
+	struct in_addr *in4_addr = NULL;
+	struct in_addr *dst_in4_addr = NULL;
+	struct net_if *iface;
+
+	iface = net_if_get_default();
+	dst_in4_addr = (struct in_addr *)net_ipv4_unspecified_address();
+	in4_addr = (struct in_addr *)
+		net_if_ipv4_select_src_addr(iface, dst_in4_addr);
```

Comment:
The iface can be set to NULL in which case the function will search all the interfaces.

So your call could look like this
```
in4_addr = net_if_ipv4_select_src_addr(NULL, net_ipv4_unspecified_address());
```

I just noticed that const is missing from net_if_ipv4_select_src_addr() dst address, I will send a patch fixing it.